fluent/migrate/transforms.py

```diff
@@ -253,23 +254,37 @@ def __call__(self, ctx):
             variant, = variants
             return evaluate(ctx, self.foreach(variant))
 
-        last_index = min(len(variants), len(keys)) - 1
-
-        def createVariant(zipped_enum):
-            index, (key, variant) = zipped_enum
+        # The default CLDR form should be the last we have in
+        # DEFAULT_ORDER, usually `other`, but in some cases `many`.
+        # If we don't have a variant for that, we'll append one,
+        # using the, in CLDR order, last existing variant in the legacy
+        # translation. That may or may not be the last variant.
+        default_key = [
+            key for key in reversed(self.DEFAULT_ORDER) if key in keys
+        ][0]
+
+        keys_and_variants = zip(keys, variants)
+        keys_and_variants.sort(key=lambda (k, v): self.DEFAULT_ORDER.index(k))
+        if keys_and_variants[-1][0] != default_key:
```

Comment:
I find the double indexes (`[-1][0]`) hard to read. I wonder if destructing the last element of `keys_and_variants` would help?

```py
last_key, last_variant = keys_and_variants[-1]
if last_key != default_key:
     keys_and_variants.append((default_key, last_variant))
```